datastore.py

```diff
@@ -128,36 +128,203 @@ def get_lap_results(self):
         self._lock.unlock()
         return temp
 
+    def get_time_at_index(self, index):
+        self._lock.lockForRead()
+        try:
+            _time = self._lap_simulation_results.time_list[index]
+        except IndexError:
+            logger.info("index out of range: {}, returning last time",
+                    extra={'sim_index':index})
+            _time = self._lap_simulation_results.time_list[-1]
+        temp = deepcopy(_time)
+        self._lock.unlock()
+        return temp
+
+    def get_time_list(self, num_index_samples):
```

Comment:
this looks like it does the same thing that `get_time_at_index()` does

Reply:
probably need line 146 to be  _time = self._lap_simulation_results.time_list[**0:**num_index_samples]